### ISSUE TITLE
Generate TLS cert for khadga.app

### ISSUE THREAD

**Issue by rarebreed** [rarebreed/khadga#27]
There's a new domain khadga.app but there needs to be a way to generate the TLS cert from LetsEncrypt.  Moreover, there needs to be a way to auto provision a new cert after the 90 day period for the TLS cert expires.

When the TLS cert expires, a new one needs to be reprovisioned and either:

- It is dynamically updated on existing pods
- New images are created with the TLS certs and then redployed

Ideally, the first option can be used.  I would hate to have to create a new docker image every 90 days.

**Comment by rarebreed** [comment]
Ran into some issues with the ingress-nginx project, which will be needed to use LetsEncrypt.

In Google Kubernetes Engine, there are 2 ways to provide external load balancing:

- LoadBalancers
- Ingress

LoadBalancers seem to be similar to AWS's network load balancers.  It is a L3 load balancer, meaning it operates at the IP level protocol.  It has no awareness of HTTP(S) traffic, only TCP and UDP.  It can however still work with TLS traffic, but requires more setup.  It's also only setup for regional load-balancing.  The external IP obtained by a LoadBalancer is reachable globally...but the services (containers) are all running within a single region.  For example, if Mei from China hits a regional external IP, she can still access it, but instead of balancing the load to a closer region, the services will still run in the us-central1-a region.  This will have latency and bandwidth issues.  For my purposes, this really isn't a huge issue, since I'm only going to have one cluster running in one region.

Ingresses are global load balancers that work at the application layer (L7).  It is aware of HTTP(S) traffic (and in fact, can only operate on http(s) traffic...not UDP for example).  Nginx is a classic example of a L7 load balancer, and indeed, this seems to be what GKE is using as its default ingress implementation.

The problem I ran into while testing the ingress-nginx load balancer was that the default settings for nginx was not properly loading my webpage.  The page was blank, and it seems as though nginx was failing to proxy back some of the files that the khadga webserver should have been sending back to the client.  For example, although the index.js was showing up when I checked in the Network tab of the developer console....the 1.js, 2.js, and the wasm files were not.

Running a vanila (the GKE default) ingress did not have this issue.  In fact, the app is now up and running on  http://http://35.241.2.197

So, there's something wrong with the ingress-nginx implementation.  As a consequence, I created a new beta feature called ManagedCertificates.  This provides you a Google generated SSL cert, albeit a limited one.  The Google generated TLS cert can not provide wildcards or SAN fields.  This means that only one CN domain name is allowed.  So while `khadga.app` will work, `www.khadga.app` will not.  This is ok I think however.

If push comes to shove, GKE does allow you to upload certificates manually.  I could then manually create LetsEncrypt certs and upload them periodically.  But that's error prone.
